labelbox/schema/bulk_import_request.py

```diff
@@ -0,0 +1,151 @@
+import json
+from pathlib import Path
+from typing import BinaryIO
+from typing import Iterable
+from typing import Tuple
+from typing import Union
+
+import requests
+
+import labelbox.exceptions
+from labelbox import Client
+from labelbox.orm import query
+from labelbox.orm.db_object import DbObject
+from labelbox.orm.model import Field
+from labelbox.orm.model import Relationship
+from labelbox.schema.enums import BulkImportRequestState
+
+NDJSON_MIME_TYPE = "application/x-ndjson"
+
+
+class BulkImportRequest(DbObject):
+    project = Relationship.ToOne("Project", False)
+    name = Field.String("name")
+    created_at = Field.DateTime("created_at")
+    created_by = Relationship.ToOne("User", False, "created_by")
+    input_file_url = Field.String("input_file_url")
+    error_file_url = Field.String("error_file_url")
+    status_file_url = Field.String("status_file_url")
+    state = Field.Enum(BulkImportRequestState, "state")
+
+    @classmethod
+    def create_from_url(
+            cls, client: Client, project_id: str, name: str,
+            url: str) -> 'BulkImportRequest':
+        query_str = """
+        mutation {
+            createBulkImportRequest(data: {
+                projectId: "%s",
+                name: "%s",
+                fileUrl: "%s"
+            }) {
+                %s
+            }
+        }
+        """ % (
+            project_id,
+            name,
+            url,
+            query.results_query_part(BulkImportRequest)
+        )
+        bulk_import_request_kwargs = client.execute(query_str)["createBulkImportRequest"]
+        return BulkImportRequest(client, bulk_import_request_kwargs)
+
+    @classmethod
+    def create_from_objects(
+            cls, client: Client, project_id: str, name: str,
+            predictions: Iterable[dict]) -> 'BulkImportRequest':
+        data_str = '\n'.join(json.dumps(prediction) for prediction in predictions)
+        data = data_str.encode('utf-8')
+        file_name = cls.__make_file_name(project_id, name)
+        request_data = cls.__make_request_data(project_id, name, file_name)
+        file_data = (file_name, data, NDJSON_MIME_TYPE)
+        response_data = cls.__send_create_file_command(
+            client, request_data, file_name, file_data)
+        return BulkImportRequest(client, response_data["createBulkImportRequest"])
+
+    @classmethod
+    def create_from_local_file(
+            cls, client: Client, project_id: str, name: str,
+            file: Path) -> 'BulkImportRequest':
+        file_name = cls.__make_file_name(project_id, name)
+        request_data = cls.__make_request_data(project_id, name, file_name)
+        with file.open('rb') as f:
```

Comment:
if we are exposing this method, let's throw in a quick ndjson.load to validate the file format?

we get enough cases where each annotation row is split across multiple lines, I think this will be a big win for cheap.